<launch>
  <arg name="urdf_path" default="/home/aliefpal/catkin_ws/src/manip2_urdf_description/urdf/manip2_urdf.xacro"/>
  <arg name="port_name" default="/dev/ttyS3"/>
  <arg name="baud" default="1000000"/>
  <arg name="read_rate" default="60"/>
  <arg name="print_reads" default="true"/>

  <param name="robot_description" command="$(find xacro)/xacro $(arg urdf_path)"/>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
    <param name="publish_frequency" value="60.0"/>
  </node>

  <node pkg="manip2_control" type="manip2_dxl_bridge.py" name="manip2_dxl_bridge" output="screen">
    <param name="port_name" value="$(arg port_name)"/>
    <param name="baud_rate" value="$(arg baud)"/>

    <param name="dxl_ids" value="[1,2,3,4,5]"/>
    <param name="dxl_id_p2" value="2"/>

    <param name="speed_p1_value" value="50"/>
    <param name="speed_p2_value" value="400"/>

    <param name="urdf_limits_deg" value="[[-90,90],[-30,40],[-110,0],[-75,75],[-90,90]]"/>
    <param name="tick_offsets_nom" value="[496,-3968,1715,508,499]"/>
    <param name="degs_per_tick_nom" value="[0.29296875,0.001185185,0.0612916,-0.29296875,0.29296875]"/>

    <param name="read_rate_hz" value="$(arg read_rate)"/>
    <param name="print_reads" value="$(arg print_reads)"/>

    <param name="home_deg" value="[0,0,0,0,0]"/>
  </node>
</launch>
